### PR TITLE
PHP7.4 as minimum required version

### PR DESCRIPTION
**Pull request by DanielBadura**
PHP7.2 is running EOL in 1 month and with this in mind I updated the minimum required PHP version. I went up to `7.4.*` as I think its not really worth to also maintain for PHP 7.3. But we can, if you want, go down to 7.3.

I also updated all dependencies to the newest minor versions and PHPUnit up to `^9.4.0`.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -26,19 +26,19 @@
         }
     ],
     "require": {
-        "php": "^7.2",
+        "php": "7.4.*",
```
Comment (DanielBadura) [inline composer.json:29]: not sure if we should already open up for php8

**Comment by DanielBadura** [comment]
I dont really understand why scrutinizer is failing. Maybe someone can give me a hint? 🦑

File: composer.json (hunk 1 above)
Comment (Ocramius) [reply composer.json:29]: Not unless we explicitly test for it

**Comment by Ocramius** [comment]
We need to upgrade phpcs, as it seems:

```
PHP Deprecated:  Array and string offset access syntax with curly braces is deprecated in /home/scrutinizer/.analysis/phpcs/vendor/squizlabs/php_codesniffer/src/Config.php on line 399
````

I don't see it being installed via composer though, but I'm OK with scrutinizer-ci failing, as it really should be slowly phased out.

File: .travis.yml [hunk 2]
```diff
@@ -6,12 +6,10 @@ git:
 matrix:
     fast_finish: true
     include:
-        - php: 7.2
-          env:
-            - COMPOSER_FLAGS="--prefer-stable --prefer-lowest"
-            - COVERAGE=1
-        - php: 7.3
         - php: 7.4
```
Comment (Ocramius) [inline .travis.yml:9]: There should be a `7.4` entry without `--prefer-stable --prefer-lowest`

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 2 above)
Comment (Ocramius) [reply .travis.yml:9]: This can now be completely scrapped, since we have GitHub actions covering all cases

Comment (DanielBadura) [reply .travis.yml:9]: Only case missing would be the nightly build for php8 as mentioned in #208 but this should not be considered here in this patch. I will rebase this today 👍

**Comment by DanielBadura** [comment]
> We need to upgrade phpcs, as it seems:
> 
> ```
> PHP Deprecated:  Array and string offset access syntax with curly braces is deprecated in /home/scrutinizer/.analysis/phpcs/vendor/squizlabs/php_codesniffer/src/Config.php on line 399
> ```
> 
> I don't see it being installed via composer though, but I'm OK with scrutinizer-ci failing, as it really should be slowly phased out.

Yes that was confusing me. It errors because of phpcs but i did not see it as a dep. I will add phpcs soon in another PR 😄

**Comment by DanielBadura** [comment]
@Ocramius this should be ready now. I rebased to get the workflows in and removed travis and appveyor - forgot it in the last patch. Also removed php7.2 and 7.3 from the ci

**Comment by DanielBadura** [comment]
Noticing that appveyor still runs on this PR but now failing. Did i miss something?

**Comment by Ocramius** [comment]
> Noticing that appveyor still runs on this PR but now failing. Did i miss something?

I would really just turn off all the other CI tools, now that we have github actions (which can test things on windows too)

**Comment by DanielBadura** [comment]
Yeah i wanted to and removed the config but its still running

**Comment by Ocramius** [comment]
@DanielBadura feel free to drop it: disabling it is up to @maglnet though, as I don't have access to repository settings.

**Comment by DanielBadura** [comment]
@Ocramius appveyor config is already dropped. I think this  patch is ready as it is.

**Comment by Ocramius** [comment]
:ship:  then :)

**Pull request merged**